isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -2178,98 +2403,253 @@ namespace isobus
+							std::uint16_t objectID = (static_cast<std::uint16_t>(data[1]) &
+							                          ((static_cast<std::uint16_t>(data[2])) << 8));
+							bool objectSelected = (0x01 == data[3]);
+							bool objectOpenForInput = true;
 
-							if (parentVT->get_connected_vt_version() >= VTVersion::Version4)
+							if (parentVT->get_vt_version_supported(VTVersion::Version4))
 							{
-								objectOpenForInput = (0x01 == (message->get_data()[4] & 0x01));
+								objectOpenForInput = (0x01 == (data[4] & 0x01));
 							}
 
-							if (VTVersion::Version6 == parentVT->get_connected_vt_version())
+							if (parentVT->get_vt_version_supported(VTVersion::Version6))
 							{
 								//! @todo process TAN
 							}
+
 							parentVT->process_select_input_object_callback(objectID, objectSelected, objectOpenForInput, parentVT);
 						}
 						break;
 
+						case static_cast<std::uint8_t>(Function::VTESCMessage):
+						{
+							std::uint16_t objectID = (static_cast<std::uint16_t>(data[1]) &
+							                          ((static_cast<std::uint16_t>(data[2])) << 8));
+							bool errorCode = data.at(3) & 0x1F;
+							if ((errorCode == static_cast<std::uint8_t>(ESCMessageErrorCode::OtherError)) ||
+							    (errorCode <= static_cast<std::uint8_t>(ESCMessageErrorCode::NoInputFieldOpen)))
+							{
+								if (parentVT->get_vt_version_supported(VTVersion::Version6))
+								{
+									//! @todo process TAN
+								}
+
+								parentVT->process_esc_message_callback(objectID, static_cast<ESCMessageErrorCode>(errorCode), parentVT);
+							}
+						}
+						break;
+
+						case static_cast<std::uint8_t>(Function::VTChangeNumericValueMessage):
+						{
+							std::uint16_t objectID = (static_cast<std::uint16_t>(data[1]) &
+							                          ((static_cast<std::uint16_t>(data[2])) << 8));
+							std::uint32_t value = (static_cast<std::uint32_t>(data[4]) &
+							                       ((static_cast<std::uint32_t>(data[5])) << 8) &
+							                       ((static_cast<std::uint32_t>(data[6])) << 16) &
+							                       ((static_cast<std::uint32_t>(data[7])) << 24));
+
+							if (parentVT->get_vt_version_supported(VTVersion::Version6))
+							{
+								//! @todo process TAN
+							}
+							parentVT->process_change_numeric_value_callback(objectID, value, parentVT);
+						}
+						break;
+
+						case static_cast<std::uint8_t>(Function::VTChangeActiveMaskMessage):
+						{
+							std::uint16_t maskObjectID = (static_cast<std::uint16_t>(data[1]) &
+							                              ((static_cast<std::uint16_t>(data[2])) << 8));
+							std::uint16_t errorObjectID = (static_cast<std::uint16_t>(data[4]) &
+							                               ((static_cast<std::uint16_t>(data[5])) << 8));
+							bool missingObjects = (0x04 == (data[3] & 0x04));
+							bool maskOrChildHasErrors = (0x02 == (data[3] & 0x08));
+							bool anyOtherError = (0x01 == (data[3] & 0x010));
```

Comment:
`(0x01 == (data[3] & 0x010)` would never be true, right? You'd have to check `(0 != (data[3] & 0x010))` or `(0x10 == (data[3] & 0x010))`. Similarly, `bool maskOrChildHasErrors = (0x02 == (data[3] & 0x08));` would always be false, because bit 4 (0x08) will never have a value of 2, so maybe double check your bit masking in some of these messages.

Reply:
Oh dang that doesn't seem right, not sure how I missed those. I'll double check them all